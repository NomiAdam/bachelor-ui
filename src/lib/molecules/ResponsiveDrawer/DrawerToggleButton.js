import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { minTablet } from '../../constants/device';
import { darkTheme, lightTheme } from '../../constants/theme';

const StyledToggleButton = styled.button`
    display: flex;
    flex-direction: column;
    justify-content: space-around;
    height: 24px;
    width: 30px;
    background: transparent;
    border: none;
    cursor: pointer;
    padding: 0;
    box-sizing: border-box;
    &:focus {
      outline: none;
    }
    @media ${ minTablet } {
        display: ${ ({ hideButton }) => (hideButton ? 'none' : 'flex') };
    }
`;

const StyledButtonLine = styled.div`
    width: 30px;
    height: 2px;
    background: ${ ({ primary }) => (primary ? darkTheme.black : lightTheme.white) };
`;

const DrawerToggleButton = ({ handleClick, mobileVersion, primary }) => (
    <StyledToggleButton className="toggle-button" onClick={ handleClick } hideButton={ mobileVersion }>
        <StyledButtonLine primary={ primary } />
        <StyledButtonLine primary={ primary } />
        <StyledButtonLine primary={ primary } />
    </StyledToggleButton>
);

DrawerToggleButton.propTypes = {
    /**
     * Function to handle onClick event
     */
    handleClick: PropTypes.func,
    /**
     * Boolean whether we should display navigation on toolbar or not
     */
    mobileVersion: PropTypes.bool,
    /**
     * Whether to use primary theme or secondary
     */
    primary: PropTypes.bool,
};

export { StyledToggleButton, StyledButtonLine };
export default DrawerToggleButton;
