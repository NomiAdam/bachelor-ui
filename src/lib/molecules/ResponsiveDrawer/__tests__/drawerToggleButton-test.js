import React from 'react';
import DrawerToggleButton, { StyledToggleButton, StyledButtonLine } from '../DrawerToggleButton';
import { shallow } from 'enzyme';

describe('DrawerToggleButton', () => {
    it('should be defined', () => {
        expect(DrawerToggleButton).toBeDefined();
    });
    it('should match the snapshot', () => {
        expect(shallow(<DrawerToggleButton />)).toMatchSnapshot();
    });
    it('should be of type StyledToggleButton', () => {
        expect(shallow(<DrawerToggleButton/>).type()).toBe(StyledToggleButton);
    });
    it('should have 3 StyledButtonLines nodes defined', () => {
       const wrapper = shallow(<DrawerToggleButton/>);
       expect(wrapper.find(StyledToggleButton).dive().find(StyledButtonLine).length).toBe(3);
    });
});
