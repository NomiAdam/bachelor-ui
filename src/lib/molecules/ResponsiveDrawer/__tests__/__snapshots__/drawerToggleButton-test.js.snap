// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DrawerToggleButton should match the snapshot 1`] = `
<DrawerToggleButton__StyledToggleButton
  className="toggle-button"
>
  <DrawerToggleButton__StyledButtonLine />
  <DrawerToggleButton__StyledButtonLine />
  <DrawerToggleButton__StyledButtonLine />
</DrawerToggleButton__StyledToggleButton>
`;
